source/common/quic/envoy_quic_connection.h

```diff
@@ -1,40 +1,21 @@
 #pragma once
 
-#if defined(__GNUC__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunused-parameter"
-#pragma GCC diagnostic ignored "-Winvalid-offsetof"
-#endif
-
-#include "quiche/quic/core/quic_connection.h"
-
-#if defined(__GNUC__)
-#pragma GCC diagnostic pop
-#endif
-
 #include <memory>
 
-#include "common/common/logger.h"
 #include "envoy/network/connection.h"
 
+#include "common/common/logger.h"
+
 namespace Envoy {
 namespace Quic {
 
 // Derived for network filter chain, stats and QoS. This is used on both client
 // and server side.
-class EnvoyQuicConnection : public quic::QuicConnection,
-                            protected Logger::Loggable<Logger::Id::connection> {
+class EnvoyQuicConnection : protected Logger::Loggable<Logger::Id::connection> {
```

Comment:
Typically, `EnvoyQuicFoo` inherits from a class in QUICHE, `EnvoyQuicConnection` no longer does it. I see you used variable name `network_connection_` elsewhere, maybe rename `EnvoyQuicConnection` to `QuicNetworkConnection`?

Reply:
done